iOS_SDK/OneSignalDevApp/OneSignalDevApp/ViewController.m

```diff
@@ -105,13 +105,29 @@ - (IBAction)logoutEmailButton:(id)sender {
     }];
 }
 
+- (IBAction)sendTagButton:(id)sender {
+    if (self.tagKey.text && self.tagKey.text.length
+        && self.tagValue.text && self.tagValue.text.length) {
+        [OneSignal sendTag:self.tagKey.text
+                     value:self.tagValue.text
+                 onSuccess:^(NSDictionary *result) {
+                     static int successes = 0;
+                     NSLog(@"successes: %d", ++successes);
+                 }
+                 onFailure:^(NSError *error) {
+                     static int failures = 0;
+                     NSLog(@"failures: %d", ++failures);
+        }];
+    }
+}
+
 - (IBAction)getTagsButton:(id)sender {
     [OneSignal getTags:^(NSDictionary *result) {
         NSLog(@"Tags: %@", result.description);
     }];
 }
 
-- (IBAction)sendTagButton:(id)sender {
+- (IBAction)sendTagsButton:(id)sender {
```

Comment:
To confirm: this original `sendTagsButton` always sent the literal string `{"key1": "value1"}` for test purposes, and now there's another option to send other key/value pairs?

Reply:
yep exactly. We have the ability to do this on Android and it will be convenient for non-devs to test with